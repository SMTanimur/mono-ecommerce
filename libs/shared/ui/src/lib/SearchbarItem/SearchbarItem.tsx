import Image from 'next/image';
import { createItemHref, getImageURL } from '@ecommerce/shared/utils/url';
import { SearchbarSearchQuery } from '@ecommerce/shared/graphql/types';
import Link from 'next/link';
import clsx from 'clsx';

export interface SearchbarItemProps {
  item: NonNullable<
    NonNullable<SearchbarSearchQuery['products']>['data'][0]['attributes']
  >;
  handleClick: () => void;
}

export const SearchbarItem: React.FC<SearchbarItemProps> = (props) => {
  const { item, handleClick } = props;

  return (
    <li className="list-none py-2" key={item.slug}>
      <Link href={createItemHref(item.slug)}>
        <a title={item.name} onClick={handleClick}>
          <div className="flex gap-2 items-center">
            <div className="bg-gray-100 h-12 w-12">
              {item.images && !!item.images?.data.length && (
                <Image
                  src={getImageURL(item.images.data[0].attributes?.url)}
                  width={48}
                  height={48}
                  objectFit="cover"
                />
              )}
            </div>
            <div className="flex flex-col justify-between">
              <span className={clsx('text-sm uppercase', 'hover:(underline)')}>
                {item.name}
              </span>
              <span className="text-xs text-gray-500">
                {item.smallDescription}
              </span>
            </div>
          </div>
        </a>
      </Link>
    </li>
  );
};
