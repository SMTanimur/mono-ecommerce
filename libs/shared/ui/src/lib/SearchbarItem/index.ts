import { SearchbarItem, SearchbarItemProps } from './SearchbarItem';

export { SearchbarItem };

export type { SearchbarItemProps };
