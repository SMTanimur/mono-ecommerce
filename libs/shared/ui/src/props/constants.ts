import { ButtonProps } from '@ecommerce/shared/ui/Button';
import {CartItemProps} from "@ecommerce/shared/ui/CartItem";
import {ProductSidebarProps} from "@ecommerce/shared/ui/ProductSidebar";
import {CartSidebarProps} from "@ecommerce/shared/ui/CartSidebar";
import {CategoryCardProps} from "@ecommerce/shared/ui/CategoryCard";
import {IconProps} from "@ecommerce/shared/ui/Icon";
import {ItemCardProps} from "@ecommerce/shared/ui/ItemCard";
import {NavButtonProps} from "@ecommerce/shared/ui/NavButton";
import {ProductGalleryProps} from "@ecommerce/shared/ui/ProductGallery";

import cat from './cat.jpg';
import cat2 from './cat2.jpeg';

console.log(cat2)

export const PRODUCT: ProductSidebarProps['product'] = {
  slug: 'some-name',
  price: 65,
  name: 'Some name',
  images: {
    data: [
      {
        attributes: {
          url: cat,
        },
      },
      {
        attributes: {
          url: cat2,
        },
      },
    ],
  },
  smallDescription: 'Small Description',
  description: 'Description',
  information: [
    {
      title: 'Title1',
      content: 'content1',
    },
  ],
};

export const ITEM: ItemCardProps['item'] = {
  slug: 'cat-cat',
  price: 65,
  name: 'cat cat',
  images: {
    data: [
      {
        attributes: {
          url: cat,
        },
      },
      {
        attributes: {
          url: cat2,
        },
      },
    ],
  },
  smallDescription: 'smallDescription',
  description: 'description',
};

export const BUTTON_DEFAULT_PROPS: ButtonProps = {
  children: 'hello',
  tag: 'button',
  size: 'base',
  type: 'primary',
  round: 'full',
};

export const CART_ITEM_DEFAULT_PROPS: CartItemProps = {
  inCartAmount: 1,
  item: PRODUCT,
  handleCartMinusClick: () => {
    console.log('cart minus');
  },
  handleCartPlusClick: () => {
    console.log('cart plus');
  },
  handleRemoveFromCart: () => {
    console.log('cart remove');
  },
  toggleFavorite: () => {
    console.log('favorite');
  },
};

export const CART_SIDEBAR_DEFAULT_PROPS: CartSidebarProps = {
  cartTotal: 100,
  isUser: true,
  handleCheckoutClick: async () => {
    console.log('checkout');
  },
  isLoading: false,
};

export const CATEGORY_CARD_DEFAULT_PROPS: CategoryCardProps = {
  category: {
    smallDescription:
      'smalll des',
    image: {
      data: {
        attributes: {
          url: cat,
        },
      },
    },
    slug: '1',
  },
};
export const ICON_DEFAULT_PROPS: IconProps = {
  icon: 'cart',
};

export const ITEM_CARD_DEFAULT_PROPS: ItemCardProps = {
  item: ITEM,
  size: 'base',
  handleCartClick: () => console.log('cart clicked'),
  handleAmountCartClick: () => console.log('amount cart clicked'),
  handleFavoriteClick: () => console.log('favorite clicked'),
};
export const NAV_BUTTON_DEFAULT_PROPS: NavButtonProps = {
  type: 'heart',
};
export const PRODUCT_GALLERY_DEFAULT_PROPS: ProductGalleryProps = {
  images: [
    {
      attributes: {
        url: cat,
      },
    },
    {
      attributes: {
        url: cat2,
      },
    },
  ],
};
