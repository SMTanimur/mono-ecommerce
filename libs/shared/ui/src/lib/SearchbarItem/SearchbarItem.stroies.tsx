import { Story, Meta } from '@storybook/react';
import { SEARCHBAR_ITEM_DEFAULT_PROPS } from '@ecommerce/shared/props/constants';
import { SearchbarItem, SearchbarItemProps } from './SearchbarItem';

export default {
  component: SearchbarItem,
  title: 'Components/Searchbar/Item',
} as Meta;

const Template: Story<SearchbarItemProps> = (args) => (
  <SearchbarItem {...args} />
);

export const Primary = Template.bind({});
Primary.args = {
  ...SEARCHBAR_ITEM_DEFAULT_PROPS,
};
