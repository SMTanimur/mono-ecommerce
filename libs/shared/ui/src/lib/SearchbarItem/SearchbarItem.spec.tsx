import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';

import { ITEM, SEARCHBAR_ITEM_DEFAULT_PROPS } from '../../props/constants';
import { SearchbarItem } from './SearchbarItem';

describe('SearchbarItem', () => {
    let element: HTMLElement;

    beforeEach(() => {
        const { baseElement } = render(
            <SearchbarItem {...SEARCHBAR_ITEM_DEFAULT_PROPS} />,
        );

        element = baseElement;
    });

    it('should render successfully', () => {
        const tree = create(
            <SearchbarItem {...SEARCHBAR_ITEM_DEFAULT_PROPS} />,
        ).toJSON();
        expect(tree).toMatchSnapshot();

        expect(element).toBeTruthy();
    });

    it('proper name and small description', () => {
        screen.getByText(ITEM.name);
        screen.getByText(ITEM.smallDescription);
    });
});
