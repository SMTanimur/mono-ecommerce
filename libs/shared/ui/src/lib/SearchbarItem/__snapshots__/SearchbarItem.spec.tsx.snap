// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchbarItem should render successfully 1`] = `
<li
  className="list-none py-2"
>
  <a
    href="/i/cat-cat"
    onClick={[Function]}
    onMouseEnter={[Function]}
    title="cat cat"
  >
    <div
      className="flex gap-2 items-center"
    >
      <div
        className="bg-gray-100 h-12 w-12"
      >
        Next image stub
      </div>
      <div
        className="flex flex-col justify-between"
      >
        <span
          className="text-sm uppercase hover:(underline)"
        >
          cat cat
        </span>
        <span
          className="text-xs text-gray-500"
        >
          smallDescription
        </span>
      </div>
    </div>
  </a>
</li>
`;
